tests/resources/dynamic_config.py

```diff
@@ -0,0 +1,33 @@
+import pandas as pd
+
+from hamilton.function_modifiers import ResolveAt, dynamic, extract_columns, parameterize, source
+
+
+@extract_columns("a", "b", "c", "d", "e")
+def df() -> pd.DataFrame:
+    """Produces a dataframe with columns a, b, c, d, e consisting entirely of 1s"""
+    return pd.DataFrame(
+        {
+            "a": [1],
+            "b": [1],
+            "c": [1],
+            "d": [1],
+            "e": [1],
+        }
+    )
+
+
+@dynamic(
+    lambda columns_to_sum_map: parameterize(
+        **{
+            key: {"col_1": source(value[0]), "col_2": source(value[1])}
+            for key, value in columns_to_sum_map.items()
+        }
+    ),
+    at=ResolveAt.COMPILE,
+)
+def generic_summation(col_1: pd.Series, col_2: pd.Series) -> pd.Series:
+    return col_1 + col_2
```

Comment:
```python
@config.resolve(
    lambda columns_to_sum_map: parameterize(
        **{
            key: {"col_1": source(value[0]), "col_2": source(value[1])}
            for key, value in columns_to_sum_map.items()
        }
    ),
    at=ResolveAt.COMPILE,
)
def generic_summation(col_1: pd.Series, col_2: pd.Series) -> pd.Series:
    return col_1 + col_2
```
I think that reads better...

Reply:
Yeah that's not bad

Reply:
OK, so I don't actually like putting this under `config` now that I think about it -- its pretty confusing. Specifically, we're not actually `injecting` anything, we're using it to delay evaluation. Furthermore, it doesn't actually have anything to do with the other `config` decorators, which people already have a tough time understanding. Its also pretty meta, so I want it kept on its own.

I'm thinking `@delay_resolution(until=CONFIG_AVAILABLE, lambda ...)`. That's very accurate what it does, lives separately, and it nice and readable.

Reply:
```python
@resolve(
    when=CONFIG_AVAILABLE,
    decorate_with=lambda: conf_var_1 -> ...):
```